reference/reference.go

```diff
@@ -236,6 +237,19 @@ func WithDigest(name Named, digest digest.Digest) (Canonical, error) {
 	}, nil
 }
 
+// Match reports whether ref matches the specified pattern.
+// See https://godoc.org/path#Match for supported patterns.
+func Match(pattern string, ref Reference) bool {
```

Comment:
Let's follow the same pattern as the stblib: this should return an error if the pattern is bad.